lightstep/resource_metric_dashboard.go

```diff
@@ -650,6 +668,125 @@ func (p *resourceUnifiedDashboardImp) setResourceDataFromUnifiedDashboard(projec
 	return nil
 }
 
+// assembleDashboardPanels takes the incoming set of UnifiedCharts which contain a mix
+// of charts and text panels, then partitions them into separate slices to
+// be processed into distinct Terraform resources.
+func assembleDashboardPanels(
+	dashboardID string,
+	chartSchemaType ChartSchemaType,
+	panels []client.UnifiedChart,
+) (
+	chartResources []interface{},
+	textPanelResources []interface{},
+	err error,
+) {
+	charts := []client.UnifiedChart{}
+	textPanels := []client.UnifiedChart{}
+
+	// Partition by type
+	for _, panel := range panels {
+		switch panel.ChartType {
+		case "timeseries":
+			charts = append(charts, panel)
+		case "text":
+			textPanels = append(textPanels, panel)
+		default:
+			return nil, nil, fmt.Errorf("unknown panel type: %s", panel.ChartType)
+		}
+	}
+
+	chartResources, err = assembleCharts(dashboardID, chartSchemaType, charts)
+	if err != nil {
+		return nil, nil, err
+	}
+	textPanelResources, err = assembleTextPanels(dashboardID, textPanels)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return chartResources, textPanelResources, nil
```

Comment:
(Adding the return value names to be explicit / more clear though I know they're not strictly required)

Reply:
I would stay away from these as they add the potential for unintended sideffectyness.